commit f352903d5d785339623a7d77a60239c76b80eb6e
Author: Bryan Whitehead <Bryan.Whitehead@microchip.com>
Date:   Mon Jan 7 14:00:09 2019 -0500

    lan743x: Remove phy_read from link status change function
    
    [ Upstream commit a0071840d2040ea1b27e5a008182b09b88defc15 ]
    
    It has been noticed that some phys do not have the registers
    required by the previous implementation.
    
    To fix this, instead of using phy_read, the required information
    is extracted from the phy_device structure.
    
    fixes: 23f0703c125b ("lan743x: Add main source files for new lan743x driver")
    Signed-off-by: Bryan Whitehead <Bryan.Whitehead@microchip.com>
    Reviewed-by: Andrew Lunn <andrew@lunn.ch>
    Signed-off-by: David S. Miller <davem@davemloft.net>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/ethernet/microchip/lan743x_main.c b/drivers/net/ethernet/microchip/lan743x_main.c
index 1393252c6e3c..42f5bfa33694 100644
--- a/drivers/net/ethernet/microchip/lan743x_main.c
+++ b/drivers/net/ethernet/microchip/lan743x_main.c
@@ -962,13 +962,10 @@ static void lan743x_phy_link_status_change(struct net_device *netdev)
 
 		memset(&ksettings, 0, sizeof(ksettings));
 		phy_ethtool_get_link_ksettings(netdev, &ksettings);
-		local_advertisement = phy_read(phydev, MII_ADVERTISE);
-		if (local_advertisement < 0)
-			return;
-
-		remote_advertisement = phy_read(phydev, MII_LPA);
-		if (remote_advertisement < 0)
-			return;
+		local_advertisement =
+			ethtool_adv_to_mii_adv_t(phydev->advertising);
+		remote_advertisement =
+			ethtool_adv_to_mii_adv_t(phydev->lp_advertising);
 
 		lan743x_phy_update_flowcontrol(adapter,
 					       ksettings.base.duplex,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1024_f352903d5d785339623a7d77a60239c76b80eb6e.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1024_f352903d5d785339623a7d77a60239c76b80eb6e.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
